packages/dd-trace/src/llmobs/telemetry.js

```diff
@@ -1,12 +1,54 @@
 'use strict'
 
+const {
+  SPAN_KIND,
+  MODEL_PROVIDER,
+  PARENT_ID_KEY,
+  SESSION_ID,
+  ROOT_PARENT_ID,
+  INTEGRATION,
+  DECORATOR
+} = require('./constants/tags')
+
 const telemetryMetrics = require('../telemetry/metrics')
+
+const LLMObsTagger = require('./tagger')
+
 const llmobsMetrics = telemetryMetrics.manager.namespace('mlobs')
 
 function incrementLLMObsSpanStartCount (tags, value = 1) {
   llmobsMetrics.count('span.start', tags).inc(value)
 }
 
+function incrementLLMObsSpanFinishedCount (span, value = 1) {
+  const mlObsTags = LLMObsTagger.tagMap.get(span)
+
+  const isRootSpan = mlObsTags[PARENT_ID_KEY] === ROOT_PARENT_ID
+  const hasSessionId = mlObsTags[SESSION_ID] != null
+  const integration = mlObsTags[INTEGRATION]
+  const autoInstrumented = integration != null
+  const decorator = mlObsTags[DECORATOR] != null
```

Comment:
nit (no action needed): i think this can be a straight `!mlObsTags[DECORATOR]`, since checking boolean values in this case we'll want to have `null/undefined` and `false` evaluate the same here.

like how we check
```python
decorator = span._get_ctx_item(DECORATOR) is True
```
instead of checking `is not None`

It would matter only if we set `mlObsTags[DECORATOR] = false` somewhere, as `false != null` evaluates to `true`. but since we either set `decorator` to `true`, or not at all (ie `null`/`undefined`), the check is fine - just more a note for consistency!

Reply:
Great, will make the change. Thanks for the suggestion!